presto-clp/src/main/java/com/facebook/presto/plugin/clp/ClpSplit.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.clp;
+
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.spi.NodeProvider;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.schedule.NodeSelectionStrategy;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.schedule.NodeSelectionStrategy.NO_PREFERENCE;
+
+public class ClpSplit
+        implements ConnectorSplit
+{
+    private final SchemaTableName schemaTableName;
+    private final String archivePath;
+    private final Optional<String> query;
+
+    @JsonCreator
+    public ClpSplit(@JsonProperty("schemaTableName") @Nullable SchemaTableName schemaTableName,
+                    @JsonProperty("archivePath") @Nullable String archivePath,
+                    @JsonProperty("query") Optional<String> query)
```

Comment:
is this query just the kqlQuery? if so, I think it might be better to use the name kqlQuery instead for better consistency

Reply:
Yes, will rename it

Reply:
Also renamed it in `ClpTableLayoutHandle`.